oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexProviderService.java

```diff
@@ -368,13 +349,19 @@ public class LuceneIndexProviderService {
     private PropertyIndexCleaner cleaner;
     private AsyncIndexesSizeStatsUpdate asyncIndexesSizeStatsUpdate;
 
+    private boolean enableCopyOnRead = true;
+    private boolean enableCopyOnWrite = true;
+
     @Activate
-    private void activate(BundleContext bundleContext, Map<String, ?> config) throws IOException {
+    private void activate(BundleContext bundleContext, ComponentContext componentContext, Configuration config) throws IOException {
```

Comment:
As said above mixing regular properties with component property types is rarely necessary, also instead of injecting the ComponentContext you can rely on a Map to inject component properties if it would be necessary (https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-activation.objects)

Reply:
Thx